compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1247,15 +1203,20 @@ impl<'tcx> TyCtxt<'tcx> {
     }
 
     pub fn allocator_kind(self) -> Option<AllocatorKind> {
```

Comment:
Should uses of this method be audited?

Reply:
Can we just turn it into a query depending on the new `resolutions` query?